packages/core/src/domain/configuration/endpointBuilder.ts

```diff
@@ -32,8 +33,15 @@ export function createEndpointBuilder(
   const buildUrlWithParameters = createEndpointUrlWithParametersBuilder(initConfiguration, endpointType)
 
   return {
-    build(api: 'xhr' | 'fetch' | 'beacon', retry?: RetryInfo) {
-      const parameters = buildEndpointParameters(initConfiguration, endpointType, configurationTags, api, retry)
+    build(api: 'xhr' | 'fetch' | 'beacon', flushReason?: FlushReason, retry?: RetryInfo) {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏In which case the `flushReason` is undefined? Could it be made required?

Reply:
In the [segment flush](https://github.com/DataDog/browser-sdk/blob/e394296baae484f64241c171b80d6220994261f1/packages/rum/src/domain/segmentCollection/segmentCollection.ts#L138), payload.flushReason is empty (because already set in the metadata).